policy-controller/k8s/index/src/index.rs

```diff
@@ -1092,65 +1092,74 @@ impl PolicyIndex {
         use authorization_policy::AuthenticationTarget;
 
         let mut identities = None;
+        let mut networks = None;
         for tgt in spec.authentications.iter() {
-            if let AuthenticationTarget::MeshTLS {
-                ref namespace,
-                ref name,
-            } = tgt
-            {
-                let namespace = namespace.as_deref().unwrap_or(&self.namespace);
-                tracing::trace!(ns = %namespace, %name, "Finding MeshTLSAuthentication");
-                if let Some(ns) = all_authentications.by_ns.get(namespace) {
-                    if let Some(authn) = ns.meshtls.get(name) {
-                        tracing::trace!(ns = %namespace, %name, ids = ?authn.matches, "Found MeshTLSAuthentication");
-                        // There can only be a single required MeshTLSAuthentication. This is
-                        // enforced by the admission controller.
-                        if identities.is_some() {
-                            bail!("policy must not include multiple MeshTLSAuthentications");
+            match tgt {
+                AuthenticationTarget::Network {
+                    ref namespace,
+                    ref name,
+                } => {
+                    let namespace = namespace.as_deref().unwrap_or(&self.namespace);
+                    tracing::trace!(ns = %namespace, %name, "Finding NetworkAuthentication");
+                    if let Some(ns) = all_authentications.by_ns.get(namespace) {
+                        if let Some(authn) = ns.network.get(name).as_ref() {
+                            tracing::trace!(ns = %namespace, %name, nets = ?authn.matches, "Found NetworkAuthentication");
```

Comment:
maybe the `namespace` and `name` fields should be added to a span?
```suggestion
                    let _span = tracing::trace_span!("network", ns = %namespace, %name).entered();
                    tracing::trace!("Finding NetworkAuthentication...");
                    if let Some(ns) = all_authentications.by_ns.get(namespace) {
                        if let Some(authn) = ns.network.get(name).as_ref() {
                            tracing::trace!(nets = ?authn.matches, "Found NetworkAuthentication");
```